### PR TITLE
Go APIView: misc fixes

### PR DESCRIPTION
**Pull request by catalinaperalta**
Fixes https://github.com/Azure/azure-sdk-for-go/issues/14343

### CODE REVIEW

**Comment by catalinaperalta** [comment]
Sample output using armstorage: https://apiview.dev/Assemblies/Review/47cd962acd2c4c0ab8b4f66f0d3ecd16

File: src/go/cmd/content.go [hunk 1]
```diff
@@ -124,19 +125,37 @@ func (c *content) parseConst(tokenList *[]Token) {
...
 // adds the specified function declaration to the exports list
 func (c *content) addFunc(pkg pkg, f *ast.FuncDecl) {
 	// create a method sig, for methods it's a combination of the receiver type
 	// with the function name e.g. "FooReceiver.Method", else just the function name.
 	sig := ""
 	if f.Recv != nil {
-		sig = "(c "
-		sig += pkg.getText(f.Recv.List[0].Type.Pos(), f.Recv.List[0].Type.End())
+		receiver := pkg.getText(f.Recv.List[0].Type.Pos(), f.Recv.List[0].Type.End())
+		if unicode.IsLower(rune(receiver[0])) || (string(receiver[0]) == "*" && unicode.IsLower(rune(receiver[1]))) {
```
Comment (jhendrixMSFT) [inline src/go/cmd/content.go:153]: I think you want [`ast.IsExported`](https://pkg.go.dev/go/ast#IsExported) here.

Comment (catalinaperalta) [reply src/go/cmd/content.go:153]: So `ast.IsExported` basically does a unicode.IsUpper underneath and from what I understood in the code it wouldnt accept receivers that have a pointer before the name...so the if would remain similarly large. I'd vote to keep it like this with the IsLower() directly being used.

Comment (jhendrixMSFT) [reply src/go/cmd/content.go:153]: Can we write our own version to take this into account then?  `ast.IsExported` also includes properly decoding the rune which we should be doing.
```go
func IsExported(name string) bool {
	ch, _ := utf8.DecodeRuneInString(name)
	return unicode.IsUpper(ch)
}
```

Comment (catalinaperalta) [reply src/go/cmd/content.go]: Done

**Review by jhendrixMSFT — APPROVED** [review]

**Pull request merged**